src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -258,32 +259,55 @@ public SeqColComparisonResultEntity compareSeqCols(
         comparisonResult.putIntoArrays("a-and-b", seqColCommonAttributes);
 
         // "elements" attribute | "total"
-        Integer seqColATotal = seqColAEntityMap.get("sequences").size();
-        Integer seqColBTotal = seqColBEntityMap.get("sequences").size();
+        Integer seqColATotal = seqColAEntityMap.get("lengths").size();
+        Integer seqColBTotal = seqColBEntityMap.get("lengths").size();
         comparisonResult.putIntoElements("total", "a", seqColATotal);
         comparisonResult.putIntoElements("total", "b", seqColBTotal);
 
         // "elements" attribute | "a-and-b"
-        List<String> commonSeqColAttributes = getCommonElementsDistinct(seqColAAttributesList, seqColBAttributesList);
-        for (String element: commonSeqColAttributes) {
+        List<String> commonSeqColAttributesValues = getCommonElementsDistinct(seqColAAttributesList, seqColBAttributesList); // eg: ["sequences", "lengths", ...]
+        for (String element: commonSeqColAttributesValues) {
             Integer commonElementsCount = getCommonElementsCount(seqColAEntityMap.get(element), seqColBEntityMap.get(element));
             comparisonResult.putIntoElements("a-and-b", element, commonElementsCount);
         }
 
         // "elements" attribute | "a-and-b-same-order"
-        for (String attribute: commonSeqColAttributes) {
+        for (String attribute: commonSeqColAttributesValues) {
             if (lessThanTwoOverlappingElements(seqColAEntityMap.get(attribute), seqColBEntityMap.get(attribute))
                     || unbalancedDuplicatesPresent(seqColAEntityMap.get(attribute), seqColBEntityMap.get(attribute))){
                 comparisonResult.putIntoElements("a-and-b-same-order", attribute, null);
             } else {
-                boolean attributeSameOrder = seqColAEntityMap.get(attribute).equals(seqColBEntityMap.get(attribute));
+                boolean attributeSameOrder = check_A_And_B_Same_Order(seqColAEntityMap.get(attribute), seqColBEntityMap.get(attribute));
                 comparisonResult.putIntoElements("a-and-b-same-order", attribute, attributeSameOrder);
             }
         }
 
         return comparisonResult;
     }
 
+    /**
+     * Check whether the array of elements of elementsA are in the same order as the ones in elementsB.
+     * Example 1:
+     *      A = ["ch1", "B", "ch2", "ch3"],
+     *      B = ["ch1", "A", "ch2", "ch3"],
+     *      Common = ["ch1", "ch2", "ch3"] # Common elements between A and B
+     *      ==> A.indexOf("ch1") == B.indexOf("ch1"), A.indexOf("ch2") == B.indexOf("ch2"), A.indexOf("ch3") == B.indexOf("ch3")
+     *      ==> Same order elements
+     *      */
+    public boolean check_A_And_B_Same_Order(List<String> elementsA, List<String> elementsB) {
```

Comment:
Suggest including [this link](https://github.com/ga4gh/seqcol-spec/blob/master/docs/decision_record.md#same-order-specification) in the comment

Reply:
Absolutely